### PR TITLE
add sizeof checks to impl declaration in tcp_socket.hpp for compatibility

### PR DESCRIPTION
**Pull request by xloem**
this is only one class, but shows the idea

personally i think it would be more maintainable and clearer to put the entire class declaration in the header file, but that's a bigger change

re https://github.com/bitshares/bitshares-fc/pull/231#issuecomment-766455020

### CODE REVIEW

File: include/fc/network/tcp_socket.hpp [hunk 1]
```diff
@@ -52,11 +55,7 @@ namespace fc {
     private:
       friend class tcp_server;
       class impl;
-      #ifdef _WIN64
-      fc::fwd<impl,0xa8> my;
-      #else
-      fc::fwd<impl,0x60> my;
-      #endif
+      fc::fwd<impl,sizeof(void* /*vtable*/)+sizeof(fc::future<size_t>)*2+sizeof(boost::asio::ip::tcp::socket)+sizeof(tcp_socket_io_hooks*)> my;
```
Comment (abitmore) [inline include/fc/network/tcp_socket.hpp]: Can you wrap the line somehow? It is too long.

**Review by abitmore — COMMENTED** [review]
Thanks!

**Comment by xloem** [comment]
wrapped.  note that I didn't do this to tcp_server, only tcp_socket .

File: include/fc/network/tcp_socket.hpp [hunk 2]
```diff
@@ -52,11 +55,14 @@ namespace fc {
...
-      #else
-      fc::fwd<impl,0x60> my;
-      #endif
+      fc::fwd<impl,
+        sizeof(void* /*vtable*/) +
+        sizeof(fc::future<size_t>) +
+        sizeof(fc::future<size_t>) +
+        sizeof(boost::asio::ip::tcp::socket) +
+        sizeof(tcp_socket_io_hooks*)
+    > my;
```
Comment (abitmore) [inline include/fc/network/tcp_socket.hpp]: Please use >= 6 whitespaces before `>` here for code alignment. :)

**Review by abitmore — COMMENTED** [review]
I don't think we need to update tcp_server. Thanks.

**Comment by xloem** [comment]
re: tcp_server: right.  I misread https://github.com/bitshares/bitshares-fc/pull/224 .

**Review by abitmore — APPROVED** [review]
Thanks!

**Pull request merged**